pkg/engine/lifecycletest/pulumi_test.go

```diff
@@ -3181,13 +3181,19 @@ func TestProtect(t *testing.T) {
 		return err
 	}
 
-	// Run a new update which will cause a replace, we should get an error
+	// Run a dry-run (preview) which will cause a replace, we should get an error.
+	// However, the preview doesn't bail, so we expect one "created" resource as a result of this operation.
 	expectedMessage = "<{%reset%}>unable to replace resource \"urn:pulumi:test::test::pkgA:m:typA::resA\"\n" +
 		"as it is currently marked for protection. To unprotect the resource, remove the `protect` flag from " +
 		"the resource in your Pulumi program and run `pulumi up`<{%reset%}>\n"
 	ins = resource.NewPropertyMapFromMap(map[string]interface{}{
 		"foo": "baz",
 	})
+	_, err = TestOp(Update).Run(project, p.GetTarget(t, snap), p.Options, true, p.BackendClient, validate)
+	assert.Error(t, err)
```

Comment:
Please don't use assert.Error, use ErrorContains or similar to check its actually the error expected.

Reply:
I thought that already happens with `expectedMessage`. I'll double check.